### PR TITLE
Truncate process.command_args

### PR DESCRIPTION
**Pull request by laurit**
Similarly to `process.command_line` we need to limit the size of `process.command_args`

### CODE REVIEW

File: custom/src/main/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabled.java [hunk 1]
```diff
@@ -57,15 +62,66 @@ public Resource apply(Resource existing, ConfigProperties config) {
...
+      }
+      return resource;
+    }
+
+    private static List<String> truncate(List<String> list) {
+      // when List is translated to String it is surrounded in [], each element is surrounded in
+      // double quotes, elements are separated with a comma
+      int maxLength = MAX_LENGTH - 4;
+      List<String> result = new ArrayList<>();
+      int size = 0;
```
Comment (mateuszrzeszutek) [inline custom/src/main/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabled.java:98]: Nit: maybe rename it to `totalLength` or sth similar?

File: custom/src/main/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabled.java [hunk 2]
```diff
@@ -57,15 +62,66 @@ public Resource apply(Resource existing, ConfigProperties config) {
...
+      }
+
+      if (existing != resource) {
+        logger.warning(
+            "Metrics are enabled. Truncating process.command_line and process.command_args resource attributes.");
+      }
+      return resource;
+    }
+
+    private static List<String> truncate(List<String> list) {
```
Comment (mateuszrzeszutek) [inline custom/src/main/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabled.java:93]: So what's the reason behind it? The length of all values summed must be <255 as well?

**Review by mateuszrzeszutek — APPROVED** [review]

File: custom/src/main/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabled.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply custom/src/main/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabled.java:93]: Can you add a comment saying that the JSON encoding of the list must be <255 chars long?

Comment (laurit) [reply custom/src/main/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabled.java:93]: done

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**